x-pack/plugins/observability/public/pages/alerts/alerts_table_t_grid.tsx

```diff
@@ -104,7 +94,6 @@ export const columns: Array<
     }),
     linkField: '*',
     id: RULE_NAME,
-    initialWidth: 400,
```

Comment:
columns that don't have an `initialWidth` automatically  use up the remaining width

Reply:
This looks good in large screen resolutions. In smaller resolutions (even at 1200px) the reason text gets cut off. 

<img width="1040" alt="Screenshot 2021-07-21 at 19 13 16" src="https://user-images.githubusercontent.com/2852703/126532327-3f6cb3e3-6fe6-440c-a49a-256e852a6910.png">

Since the table is not responsive, could we possibly add an `overflow: scroll` to the EventsTBody as shown here https://github.com/elastic/kibana/pull/105446/commits/8bb0f07bb68528a438ef35430a229f75321d28bc#diff-d98503225da4a83c1a846a2185b182229b868fd580d0ef19182ff235e5f98a41R206? At least there will be a horizontal scrollbar and user will be able to read the text. 

Unless there is a responsive attribute that we could add to the EuiDataGrid Table that automatically makes the table responsive? Are you aware of something like this?

Reply:
@andrew-goldstein For bigger screen resolutions that looks good! But in screen resolutions 1200 and below the text gets cut off. Since table is currently not responsive, I suggest we add an overflow: auto to 
<img width="1040" alt="Screenshot 2021-07-21 at 19 13 16" src="https://user-images.githubusercontent.com/2852703/126531108-3aff6b92-26d4-4123-9851-fe51933fae69.png">

<img width="1040" alt="Screenshot 2021-07-21 at 19 13 16" src="https://user-images.githubusercontent.com/2852703/126530998-a0d806b8-7e34-4128-8112-eed8f7906fa6.png">

Reply:
The table itself is responsive, and after removing `initialWidth`, the `Reason` column expands to fill with width of the browser, per the animated gif below:

![responsive](https://user-images.githubusercontent.com/4459398/126712396-ca0348c7-c810-4287-9803-9bb9b6e0859b.gif)

The `...` truncation was not being applied to the link text at smaller resolutions due to a subtle change in rendering behavior when `EuiLink` is used with an `onClick` prop:

`EuiDataGrid` automatically applies text truncation to the cell, but `EuiLink` automatically renders links via a `button` instead of an `a` when an `onClick` prop is passed to `EuiLink`.

I'm pushing a commit with the following change in `render_cell_value.tsx`, but I won't merge it until I hear back from @chandlerprall to see if there's another way to achieve this without passing an empty `href` to `EuiLInk`:

```ts
          // NOTE: EuiLink automatically renders links using a <button>
          // instead of an <a> when an `onClick` prop is provided, but this
          // breaks text-truncation in `EuiDataGrid`, because (per the HTML
          // spec), buttons are *always* rendered as `inline-block`, even if
          // `display` is overridden. Passing an empty `href` prop forces
          // `EuiLink` to render the link as an (inline) <a>, which enables
          // text truncation, but requires overriding the linter warning below:
          // eslint-disable-next-line @elastic/eui/href-or-on-click
          <EuiLink href="" onClick={() => setFlyoutAlert && setFlyoutAlert(alert)}>
            {alert.reason}
          </EuiLink>
```

Reply:
> I'm pushing a commit with the following change in `render_cell_value.tsx`, but I won't merge it until I hear back from @chandlerprall to see if there's another way to achieve this without passing an empty `href` to `EuiLInk`:

Per an out-of-band conversation with @chandlerprall, we decided to merge this PR as-is, and follow-up re: the empty `href` on `EuiLink` in a separate PR, if necessary.

Reply:
Post-merge update: this workaround is valid for now, and tracked by the following `EUI` issue:  https://github.com/elastic/eui/issues/4990 

Details from the issue above:

> Apparently by HTML's design / spec, text inside a button element will not receive a parent's text-overflow: ellipsis - https://stackoverflow.com/questions/9905409/is-it-possible-to-use-text-overflow-ellipsis-on-a-button-element